Gems/Atom/Feature/Common/Code/Source/ProfilingCaptureSystemComponent.cpp

```diff
@@ -603,55 +603,49 @@ namespace AZ
+                const bool captureEnded = AZ::RHI::CpuProfiler::Get()->EndContinuousCapture(captureResult);
+                if (!captureEnded)
+                {
+                    AZ_TracePrintf("ProfilingCaptureSystemComponent", "Could not end the continuous capture, is one in progress?\n");
+                    m_cpuDataSerializationInProgress.store(false);
+                    return false;
+                }
 
-            AZStd::ring_buffer<RHI::CpuProfiler::TimeRegionMap> captureResult;
-            const bool captureEnded = AZ::RHI::CpuProfiler::Get()->EndContinuousCapture(captureResult);
-            if (!captureEnded)
-            {
-                AZ_TracePrintf("ProfilingCaptureSystemComponent", "Could not end the continuous capture, is one in progress?\n");
-                return false;
-            }
+                // cpuProfilingData could be 1GB+ once saved, so use an IO thread to write it to disk.
+                auto threadIoFunction =
+                    [data = AZStd::move(captureResult), filePath = AZStd::string(outputFilePath), &flag = m_cpuDataSerializationInProgress]()
+                {
+                    SerializeCpuProfilingData(data, filePath, true);
+                    flag.store(false);
+                };
+                
+                // If the thread object already exists (ex. we have already serialized data), join. This will not block since
+                // m_cpuDataSerializationInProgress was false, meaning the IO thread has already completed execution.
+                // TODO Use a reusable thread implementation over repeated creation + destruction of threads [ATOM-16214]
+                if (m_cpuDataSerializationThread.joinable())
+                {
+                    m_cpuDataSerializationThread.join();
+                }
 
-            // cpuProfilingData could be 1GB+ once saved, so use an IO thread to write it to disk.
-            m_cpuDataSerializationInProgress.store(true);
-            auto threadIoFunction =
-                [data = AZStd::move(captureResult), filePath = AZStd::string(outputFilePath), &flag = m_cpuDataSerializationInProgress]()
-            {
-                SerializeCpuProfilingData(data, filePath, true);
-                flag.store(false);
-            };
-            
-            // If the thread object already exists (ex. we have already serialized data), join. This will not block since
-            // m_cpuDataSerializationInProgress is false, meaning the IO thread has already completed execution.
-            // TODO Use a reusable thread implementation over repeated creation + destruction of threads [ATOM-16214]
-            if (m_cpuDataSerializationThread.joinable())
-            {
-                m_cpuDataSerializationThread.join();
-            }
+                auto thread = AZStd::thread(threadIoFunction);
+                m_cpuDataSerializationThread = AZStd::move(thread);
```

Comment:
Random nit, you can just do `m_cpuDataSerializationThread = AZStd::thread{threadIoFunction}` here.